wrappers/services.go

```diff
@@ -75,30 +76,109 @@ func generateSnapServiceFile(app *snap.AppInfo, opts *AddSnapServicesOptions) ([
 	return genServiceFile(app, opts)
 }
 
+// max returns the maximum of two integers. Why is this
+// not provided by golang?
+func max(a, b int) int {
+	if a < b {
+		return b
+	}
+	return a
+}
+
+func min(a, b int) int {
+	if a > b {
+		return b
+	}
+	return a
+}
+
+func formatCpuGroupSlice(grp *quota.Group) string {
+	if grp.CpuLimit == nil {
+		return ""
+	}
+
+	// calculateSystemdCPULimit calculates the number of cpus and the allowed percentage
+	// to the systemd specific format. The CPUQuota setting is only available since systemd 213
+	calculateSystemdCPULimit := func() string {
+		if grp.CpuLimit.Count == 0 && grp.CpuLimit.Percentage == 0 {
+			return ""
+		}
+
+		cpuQuotaFormat := "CPUQuota=%d%%\n"
```

Comment:
nitpick: I wouldn't bother declaring these as variables, generally the format string is passed literally to the Sprintf function

Reply:
Thanks, good point!